# Copyright 2007 Sun Microsystems, Inc.  All rights reserved.
# Use subject to license terms.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, and/or sell copies of the Software, and to permit persons
# to whom the Software is furnished to do so, provided that the above
# copyright notice(s) and this permission notice appear in all copies of
# the Software and that both the above copyright notice(s) and this
# permission notice appear in supporting documentation.
# 
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
# OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
# HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
# INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
# FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
# NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
# WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
# 
# Except as contained in this notice, the name of a copyright holder
# shall not be used in advertising or otherwise to promote the sale, use
# or other dealings in this Software without prior written authorization
# of the copyright holder.

Add print message from VBE probing.
Use new function xf86ModesAdd() to add mode.
diff -urp -x '*~' src/nv_driver.c src/nv_driver.c
--- src/nv_driver.c	2007-01-02 15:15:58.106491000 -0800
+++ src/nv_driver.c	2007-01-02 15:15:58.691509000 -0800
@@ -1538,9 +1538,7 @@ NVPreInit(ScrnInfoPtr pScrn, int flags)
         Mode = xf86CVTMode(pNv->fpWidth, pNv->fpHeight, 60.00, TRUE, FALSE);
 	Mode->type = M_T_BUILTIN;
 	Mode->HSync = 0;
-	Mode->next = pScrn->monitor->Modes;
-	pScrn->monitor->Modes->prev = Mode;
-	pScrn->monitor->Modes = Mode;
+	pScrn->monitor->Modes = xf86ModesAdd(pScrn->monitor->Modes, Mode);
     }
 
     /*
diff -urp -x '*~' src/nv_setup.c src/nv_setup.c
--- src/nv_setup.c	2007-01-02 15:15:58.107400000 -0800
+++ src/nv_setup.c	2007-01-02 15:15:58.692535000 -0800
@@ -476,10 +476,12 @@ NVCommonSetup(ScrnInfoPtr pScrn)
 
     if(!pNv->twoHeads) {
        pNv->CRTCnumber = 0;
-       if(!(monitorA = NVProbeDDC(pScrn, 0)))
+       if(!(monitorA = NVProbeDDC(pScrn, 0))) {
               monitorA = nvProbeDDC(pScrn, 
               	xf86GetEntityInfo(pScrn->entityList[0])->index);
-
+	      if (monitorA)
+       		xf86PrintEDID(monitorA);
+       }
        if (monitorA) {
            FlatPanel = monitorA->features.input_type ? 1 : 0;
 
@@ -567,9 +569,12 @@ NVCommonSetup(ScrnInfoPtr pScrn)
        monitorA = NVProbeDDC(pScrn, 0);
        monitorB = NVProbeDDC(pScrn, 1);
 
-       if (!monitorA && !monitorB)
+       if (!monitorA && !monitorB) {
               monitorA = nvProbeDDC(pScrn, 
               	xf86GetEntityInfo(pScrn->entityList[0])->index);
+	      if (monitorA)
+       		xf86PrintEDID(monitorA);
+       }
 
        if(slaved_on_A && !tvA) {
           CRTCnumber = 0;
